diff --git a/ThirdParty/Minizip/minizip/ioapi.h b/ThirdParty/Minizip/minizip/ioapi.h
index 8dcbdb0..493234c 100644
--- a/ThirdParty/Minizip/minizip/ioapi.h
+++ b/ThirdParty/Minizip/minizip/ioapi.h
@@ -23,8 +23,9 @@
 
 #if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__))
 
-  // Linux needs this to support file operation on files larger then 4+GB
-  // But might need better if/def to select just the platforms that needs them.
+  /* Linux needs this to support file operation on files larger then 4+GB
+     But might need better if/def to select just the platforms that needs them.
+  */
 
         #ifndef __USE_FILE_OFFSET64
                 #define __USE_FILE_OFFSET64
@@ -186,8 +187,9 @@ typedef struct zlib_filefunc64_32_def_s
 
 #define ZREAD64(filefunc,filestream,buf,size)     ((*((filefunc).zfile_func64.zread_file))   ((filefunc).zfile_func64.opaque,filestream,buf,size))
 #define ZWRITE64(filefunc,filestream,buf,size)    ((*((filefunc).zfile_func64.zwrite_file))  ((filefunc).zfile_func64.opaque,filestream,buf,size))
-//#define ZTELL64(filefunc,filestream)            ((*((filefunc).ztell64_file)) ((filefunc).opaque,filestream))
-//#define ZSEEK64(filefunc,filestream,pos,mode)   ((*((filefunc).zseek64_file)) ((filefunc).opaque,filestream,pos,mode))
+/* #define ZTELL64(filefunc,filestream)            ((*((filefunc).ztell64_file)) ((filefunc).opaque,filestream))
+   #define ZSEEK64(filefunc,filestream,pos,mode)   ((*((filefunc).zseek64_file)) ((filefunc).opaque,filestream,pos,mode))
+*/
 #define ZCLOSE64(filefunc,filestream)             ((*((filefunc).zfile_func64.zclose_file))  ((filefunc).zfile_func64.opaque,filestream))
 #define ZERROR64(filefunc,filestream)             ((*((filefunc).zfile_func64.zerror_file))  ((filefunc).zfile_func64.opaque,filestream))
 
diff --git a/src/ZIP/src/fmi_zip_miniunz.c b/src/ZIP/src/fmi_zip_miniunz.c
index 1931161..33ddc1a 100644
--- a/src/ZIP/src/fmi_zip_miniunz.c
+++ b/src/ZIP/src/fmi_zip_miniunz.c
@@ -142,7 +142,7 @@ int mymkdir(dirname)
 #elif __APPLE__
     ret = mkdir (dirname,0775);
 #else
-#error Unknown platform
+    ret = mkdir (dirname,0775);
 #endif
     return ret;
 }
