diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e90fe40..c6b9d70 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -272,6 +272,8 @@ if (USE_XFT)
     endif (LIB_fontconfig)
 endif (USE_XFT)
 
+if (NOT OPTION_BUILD_SHARED_LIBS)
+
 #######################################################################
 
 FL_ADD_LIBRARY(fltk STATIC "${STATIC_FILES}")
@@ -323,45 +325,45 @@ endif (OPENGL_FOUND)
 # at least we can build shared libraries (dlls) with Visual Studio.
 # AlbrechtS
 
-if (OPTION_BUILD_SHARED_LIBS)
+else()
 
     FL_ADD_LIBRARY(fltk SHARED "${SHARED_FILES}")
-    target_link_libraries(fltk_SHARED ${OPTIONAL_LIBS})
+    target_link_libraries(fltk ${OPTIONAL_LIBS})
 
     ###################################################################
 
     FL_ADD_LIBRARY(fltk_forms SHARED "${FLCPPFILES}")
     if (MSVC)
-      target_link_libraries(fltk_forms_SHARED fltk)
+      target_link_libraries(fltk_forms fltk)
     else ()
-      target_link_libraries(fltk_forms_SHARED fltk_SHARED)
+      target_link_libraries(fltk_forms fltk)
     endif (MSVC)
 
     ###################################################################
 
     FL_ADD_LIBRARY(fltk_images SHARED "${IMGCPPFILES}")
     if (MSVC)
-      target_link_libraries(fltk_images_SHARED fltk)
+      target_link_libraries(fltk_images fltk)
     else ()
-      target_link_libraries(fltk_images_SHARED fltk_SHARED)
+      target_link_libraries(fltk_images fltk)
     endif (MSVC)
 
     if (OPTION_USE_SYSTEM_LIBJPEG)
-	target_link_libraries(fltk_images_SHARED ${FLTK_JPEG_LIBRARIES})
+	target_link_libraries(fltk_images ${FLTK_JPEG_LIBRARIES})
     else()
-	target_link_libraries(fltk_images_SHARED fltk_jpeg_SHARED)
+	target_link_libraries(fltk_images fltk_jpeg)
     endif (OPTION_USE_SYSTEM_LIBJPEG)
 
     if (OPTION_USE_SYSTEM_LIBPNG)
-	target_link_libraries(fltk_images_SHARED ${FLTK_PNG_LIBRARIES})
+	target_link_libraries(fltk_images ${FLTK_PNG_LIBRARIES})
     else()
-	target_link_libraries(fltk_images_SHARED fltk_png_SHARED)
+	target_link_libraries(fltk_images fltk_png)
     endif (OPTION_USE_SYSTEM_LIBPNG)
 
     if (OPTION_USE_SYSTEM_ZLIB)
-	target_link_libraries(fltk_images_SHARED ${FLTK_ZLIB_LIBRARIES})
+	target_link_libraries(fltk_images ${FLTK_ZLIB_LIBRARIES})
     else()
-	target_link_libraries(fltk_images_SHARED fltk_z_SHARED)
+	target_link_libraries(fltk_images fltk_z)
     endif (OPTION_USE_SYSTEM_ZLIB)
 
     ###################################################################
@@ -369,9 +371,9 @@ if (OPTION_BUILD_SHARED_LIBS)
     if (OPENGL_FOUND)
       FL_ADD_LIBRARY(fltk_gl SHARED "${GLCPPFILES}")
       if (MSVC)
-	target_link_libraries(fltk_gl_SHARED fltk ${OPENGL_LIBRARIES})
+	target_link_libraries(fltk_gl fltk ${OPENGL_LIBRARIES})
       else ()
-	target_link_libraries(fltk_gl_SHARED fltk_SHARED ${OPENGL_LIBRARIES})
+	target_link_libraries(fltk_gl fltk ${OPENGL_LIBRARIES})
       endif (MSVC)
     endif (OPENGL_FOUND)
 
